PYME/LMVis/pipeline.py

```diff
@@ -759,63 +762,46 @@ def OpenFile(self, filename= '', ds = None, **kwargs):
 
-            
-        #wrap the data source with a mapping so we can fiddle with things
-        #e.g. combining z position and focus 
-        mapped_ds = tabular.MappingFilter(ds)
-
-        
-        if 'PixelSize' in kwargs.keys():
-            mapped_ds.addVariable('pixelSize', kwargs['PixelSize'])
-            mapped_ds.setMapping('x', 'x*pixelSize')
-            mapped_ds.setMapping('y', 'y*pixelSize')
-
-        #extract information from any events
-        self.ev_mappings, self.eventCharts = _processEvents(mapped_ds, self.events, self.mdh)
+        # skip the MappingFilter wrapping, etc. in self.addDataSource and add this datasource as-is
+        ds.mdh, ds.events = self.mdh, self.events
+        self.dataSources['results_source'] = ds
 
+        add_pipeline_variables = Pipelineify(self.recipe, 
+            inputFitResults='results_source',
+            inputDriftResults='' if self.driftInputMapping is None else self.driftInputMapping,
+            inputEvents='' if self.events is None else self.events,
+            pixelSizeNM=1. if 'PixelSize' not in kwargs.keys() else kwargs['PixelSize'],
+            outputLocalizations='Localizations')
+        self.recipe.add_module(add_pipeline_variables)
 
+        # FIXME - we do this already in pipelinify, maybe we can avoid doubling up?
+        self.ev_mappings, self.eventCharts = _processEvents(ds, self.events, self.mdh)  # extract information from any events
 
         #Fit module specific filter settings        
         if 'Analysis.FitModule' in self.mdh.getEntryNames():
             fitModule = self.mdh['Analysis.FitModule']
-            
-            #print 'fitModule = %s' % fitModule
-            
             if 'Interp' in fitModule:
                 self.filterKeys['A'] = (5, 100000)
-            
-            if 'LatGaussFitFR' in fitModule:
-                mapped_ds.addColumn('nPhotons', getPhotonNums(mapped_ds, self.mdh))
-
-            if 'SplitterFitFNR' in fitModule:
-                mapped_ds.addColumn('nPhotonsg', getPhotonNums({'A': mapped_ds['fitResults_Ag'], 'sig': mapped_ds['fitResults_sigma']}, self.mdh))
-                mapped_ds.addColumn('nPhotonsr', getPhotonNums({'A': mapped_ds['fitResults_Ar'], 'sig': mapped_ds['fitResults_sigma']}, self.mdh))
-                mapped_ds.setMapping('nPhotons', 'nPhotonsg+nPhotonsr')
-
             if fitModule == 'SplitterShiftEstFR':
                 self.filterKeys['fitError_dx'] = (0,10)
                 self.filterKeys['fitError_dy'] = (0,10)
 
         #self._get_dye_ratios_from_metadata()
 
-        self.addDataSource('Localizations', mapped_ds)
-
         # Retrieve or estimate image bounds
-        if False:  # 'imgBounds' in kwargs.keys():
-            self.imageBounds = kwargs['imgBounds']
-        elif (not (
-                'scanx' in mapped_ds.keys() or 'scany' in mapped_ds.keys())) and 'Camera.ROIWidth' in self.mdh.getEntryNames():
+        # if False:  # 'imgBounds' in kwargs.keys():
```

Comment:
```suggestion
        if False:  # 'imgBounds' in kwargs.keys():
```

Reply:
I think you're right that the real-time display is the only situation where that is currently passed (and ignored). Also worth noting that the kwarg is misspelled there (`imBound`) in LMAnalysis, line 725.

I'm completely fine with whatever you decide to do here, @David-Baddeley. I'm a bit new to the suggestion feature - would you like me to commit it as I'm OK with it?

Reply:
I'll commit them from this end - I think that we might ultimately want to pop this out as an option in the matlab and .txt loaders as it would make it easier to get raw images /overviews taken in other software to line up with rendered images.